velox/connectors/hive/HiveConnector.h

```diff
@@ -92,38 +97,119 @@ class HiveTableHandle : public ConnectorTableHandle {
   const core::TypedExprPtr remainingFilter_;
 };
 
+/**
+ * Location related properties of the Hive table to be written
+ */
+class LocationHandle {
+ public:
+  enum class TableType { kNew, kExisting, kTemporary };
+  enum class WriteMode {
+    kStageAndMoveToTargetDirectory,
+    kDirectToTargetNewDirectory,
+    kDirectToTargetExistingDirectory
+  };
+
+  LocationHandle(
+      const std::string& targetPath,
+      const std::string& writePath,
+      TableType tableType,
+      WriteMode writeMode)
+      : targetPath_(targetPath),
+        writePath_(writePath),
+        tableType_(tableType),
+        writeMode_(writeMode) {}
+
+  const std::string& targetPath() const {
+    return targetPath_;
+  }
+
+  const std::string& writePath() const {
+    return writePath_;
+  }
+
+  TableType tableType() const {
+    return tableType_;
+  }
+
+  WriteMode writeMode() const {
+    return writeMode_;
+  }
+
+ private:
+  const std::string targetPath_;
+  const std::string writePath_;
+  const TableType tableType_;
+  const WriteMode writeMode_;
+};
+
 /**
  * Represents a request for Hive write
  */
 class HiveInsertTableHandle : public ConnectorInsertTableHandle {
  public:
-  explicit HiveInsertTableHandle(const std::string& filePath)
-      : filePath_(filePath) {}
+  explicit HiveInsertTableHandle(
+      std::vector<std::shared_ptr<const HiveColumnHandle>>& inputColumns,
+      std::shared_ptr<const LocationHandle> locationHandle,
+      bool isCreateTable)
```

Comment:
Why do we need this boolean? Can we not deduce it from LocationHandle::tableType == kNew?